### ISSUE TITLE
iOS accessibility bridge does not detect scope route change correctly

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#66308]
## Steps to Reproduce

```dart
import 'package:flutter/material.dart';
final GlobalKey<NavigatorState> navigator = GlobalKey<NavigatorState>();
void main() {
  runApp(MaterialApp(home:TapBox()));
}

class TapBox extends StatelessWidget {
  TapBox({Key key}) : super(key: key);
 
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        actions: [
          // action button
          IconButton(icon: const BackButtonIcon(), onPressed: () {}),
        ],
      ),
      body: TapboxA(),
    );
  }
}
 
class TapboxA extends StatefulWidget {
  TapboxA({Key key}) : super(key: key);
 
  @override
  _TapboxAState createState() => _TapboxAState();
}
 
class _TapboxAState extends State<TapboxA> {
  bool _active = false;
 
  final pageViewKey = GlobalKey();
  final pageController = PageController(initialPage: 0, viewportFraction: 0.15);
 
  void _handleTap() {
    setState(() {
      _active = !_active;
    });
  }
 
  @override
  Widget build(BuildContext context) {
    return _active
        ? _buildPageView(
      context,
    )
        : Semantics(
            key: ValueKey<int>(1),
            scopesRoute: true,
            explicitChildNodes: true,
            child: Semantics(
              namesRoute: true,
              label: 'PageView widget',
              onTap: _handleTap,
              excludeSemantics: true,
              child: _buildPageView(
                context,
              ),
            )
          );
  }
 
  Widget _buildPageView(BuildContext context) {
    return Semantics(
      key: ValueKey<int>(0),
      scopesRoute: true,
      explicitChildNodes: true,
      child: PageView.builder(
        key: pageViewKey,
        controller: pageController,
        allowImplicitScrolling: true,
        itemCount: 20,
        itemBuilder: (_, index) => _buildItem(
          context,
          index: index,
        ),
      )
    );
  }
 
  Widget _buildItem(
      BuildContext context, {
        int index,
      }) {
    return Semantics(
      label: 'label',
      namesRoute: true,
      button: true,
      excludeSemantics: true,
      onTap: () async {
        await _animateToPage(index);
        _handleTap();
      },
      child: Center(child: const Text('text')),
    );
  }
 
  Future<void> _animateToPage(int page) {
    return pageController.animateToPage(
      page,
      duration: const Duration(milliseconds: 400),
      curve: Curves.easeInOut,
    );
  }
}
```

1. iOS voice over on
2. tap the pageview widget
3. double tap to trigger on tap event
4, the focus go to the first text
5. double tap to trigger on tap

expect: it should announce `pageview widget`
action: nothing is announced

**Comment by mehmetf** [comment]
This became a release blocker for `customer:money`.

**Comment by chunhtai** [comment]
pr is good to go, but tree is close in preparing for a big migration which might take one or two days. I will land the pr right after that
